### PR TITLE
Fix NMS Return Index

### PR DESCRIPTION
**Pull request by huayuan4396**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by grimoire** [comment]
Please describe the motivation of this PR

**Comment by huayuan4396** [comment]
> Please describe the motivation of this PR

I think the returned index should be the selected index of the initial input boxes, considering pre_top_k, nms, and keep_top_k. But it seems that the current code does not include the index transformation of nms selection.

File: mmdeploy/mmcv/ops/nms.py [hunk 1]
```diff
@@ -345,7 +345,8 @@ def _multiclass_nms_single(boxes: Tensor,
     labels = labels[:, topk_inds, ...]
 
     if output_index:
-        bbox_index = pre_topk_inds[None, topk_inds]
+        bbox_index = pre_topk_inds[box_inds]
+        bbox_index = bbox_index[None, topk_inds[:-1]]
```
Comment (grimoire) [inline mmdeploy/mmcv/ops/nms.py:349]: what's `[:-1]` for?

File: mmdeploy/mmcv/ops/nms.py [hunk 2]
```diff
@@ -345,7 +345,8 @@ def _multiclass_nms_single(boxes: Tensor,
     labels = labels[:, topk_inds, ...]
 
     if output_index:
-        bbox_index = pre_topk_inds[None, topk_inds]
+        bbox_index = pre_topk_inds[box_inds]
```
Comment (grimoire) [inline mmdeploy/mmcv/ops/nms.py:348]: pre_topk_inds could be None if pre_top_k > 0.

File: mmdeploy/mmcv/ops/nms.py (hunk 1 above)
Comment (huayuan4396) [reply mmdeploy/mmcv/ops/nms.py:349]: https://github.com/open-mmlab/mmdeploy/blob/90f17f2e62ef3079dfcdb21c364892a43a2ff210/mmdeploy/mmcv/ops/nms.py#L332-L334
There's a pad here, would be out of bound without [:-1].

File: mmdeploy/mmcv/ops/nms.py (hunk 2 above)
Comment (huayuan4396) [reply mmdeploy/mmcv/ops/nms.py:348]: Have fixed for pre_top_k<0 or keep_top_k<0.

**Review by grimoire — APPROVED** [review]
LGTM

**Review by RunningLeon — APPROVED** [review]
LGTM

**Pull request merged**